File: string/1736_latest_time_by_replacing_hidden_digits/1736_latest_time_by_replacing_hidden_digits.py
class Solution:
    def maximumTime(self, time: str) -> str:
        hh = time[:2]
        if hh[0] == '?' and hh[1] == '?':
            hh = '23'
        elif hh[0] == '?':
            if hh[1] < '4':
                hh = f'2{hh[1]}'
            else:
                hh = f'1{hh[1]}' 
        elif hh[1] == '?':
            if hh[0] == '2':
                hh = f'{hh[0]}3'
            else:
                hh = f'{hh[0]}9'
        mm = time[3:]
        if mm[0] == '?':
            mm = f'5{mm[1]}'
        if mm[1] == '?':
            mm = f'{mm[0]}9'
        return f'{hh}:{mm}'


time = "2?:?0"
res = Solution().maximumTime(time)
print(res)

File: string/1736_latest_time_by_replacing_hidden_digits/1736_latest_time_by_replacing_hidden_digits.go
package main

import "fmt"

func maximumTime(time string) string {
	times := []byte(time)
	if times[0] == '?' {
		if times[1] == '?' || times[1] < '4' {
			times[0] = '2'
		} else {
			times[0] = '1'
		}
	}
	if times[1] == '?' {
		if times[0] == '2' {
			times[1] = '3'
		} else {
			times[1] = '9'
		}
	}
	if times[3] == '?' {
		times[3] = '5'
	}
	if times[4] == '?' {
		times[4] = '9'
	}
	return string(times)
}

func main() {
	time := "2?:?0"
	res := maximumTime(time)
	fmt.Println(res)
}
